alkira/resource_alkira_checkpoint_helper.go

```diff
@@ -0,0 +1,194 @@
+package alkira
+
+import (
+	"errors"
+	"log"
+	"strconv"
+
+	"github.com/alkiranet/alkira-client-go/alkira"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type CheckpointGetSegById func(string) (alkira.Segment, error)
+
+func expandCheckpointInstances(in *schema.Set) []alkira.CheckpointInstance {
+	if in == nil || in.Len() == 0 {
+		log.Printf("[DEBUG] invalid Checkpoint instance input")
+		return nil
+	}
+
+	instances := make([]alkira.CheckpointInstance, in.Len())
+	for i, instance := range in.List() {
+		r := alkira.CheckpointInstance{}
+		instanceCfg := instance.(map[string]interface{})
+		if v, ok := instanceCfg["name"].(string); ok {
+			r.Name = v
+		}
+		if v, ok := instanceCfg["credential_id"].(string); ok {
+			r.CredentialId = v
+		}
+		instances[i] = r
+	}
+
+	return instances
+}
+
+func expandCheckpointSegmentOptions(in *schema.Set, fn CheckpointGetSegById) (map[string]alkira.OuterZoneToGroups, error) {
```

Comment:
what's the point to have two functions? expand is pretty much same as convert... Could we combine two.